src/engine/Game.java

```diff
@@ -95,43 +85,71 @@ private void zombieTurn() {
 				int[] newCoordinates = board.moveUnit(((Zombie) key).getSpeed(), coordinates[0], coordinates[1]);
 				if(newCoordinates == null) //if the zombies get to the end
 				{
-					endGame(false);
+					endGame(false); //zombies win
 					return;
 				}
 				board.getExperimental().replace(key, coordinates, newCoordinates);
 			}
 		}
 		
-		Random rand = new Random();
-		int rowNumber = rand.nextInt(levelInfo.getGridY()); //determines which row the zombie will go down
 		
-		for(int i = 0; i < 1; i++)  //spawn zombies   1 can replaced with a variable later
-		{
-			Regular_Zombie zombie = new Regular_Zombie();
-			board.addUnit(zombie, rowNumber, levelInfo.getGridX() - 1);
-		}
+		//spawn new zombies
+		if (!zombieQueue.isEmpty()) { //there must be zombies to spawn
+			Random rand = new Random();
+			int zombiesToSpawn = rand.nextInt(numZombies/4);
+			
+			if (zombiesToSpawn > zombieQueue.values().stream().mapToInt(Integer::intValue).sum()) { 
+				//if the random number is larger than the reamining zombies then spawn all remaining zombies
+				zombiesToSpawn = zombieQueue.values().stream().mapToInt(Integer::intValue).sum();
+			}
+			
+			LOG.debug("Spawning " + zombiesToSpawn + " zombies");
+			
+			for(int i = 0; i < zombiesToSpawn; i++)  //spawn zombies
+			{
+				//determine type of zombie spawn
+				List<ZombieTypes> keys = new ArrayList<ZombieTypes>(zombieQueue.keySet());

```

Comment:
You can use enum class `values()` method (i.e., `ZombieType.value()`) to return a ZombieType array of the all of the different enum  values.

Reply:
I see thanks for clarifying.